controllers/machine_controller.go

```diff
@@ -465,7 +465,7 @@ func (r *MachineReconciler) reconcileDeleteExternal(ctx context.Context, m *clus
 }
 
 func (r *MachineReconciler) shouldAdopt(m *clusterv1.Machine) bool {
-	return !util.HasOwner(m.OwnerReferences, clusterv1.GroupVersion.String(), []string{"MachineSet", "Cluster"})
+	return metav1.GetControllerOf(m) == nil && !util.HasOwner(m.OwnerReferences, clusterv1.GroupVersion.String(), []string{"MachineSet", "Cluster"})
```

Comment:
This is probably worth adding a comment (that an example of a controller owner would be a control plane resource), as well as a unit test?

Reply:
yeah, maybe I can also remove the MachineSet from the list on the right, given that it should already be a controller reference